server/routerlicious/packages/lambdas/src/broadcaster/lambda.ts

```diff
@@ -78,26 +78,23 @@ export class BroadcasterLambda implements IPartitionLambda {
     }
 
     private sendPending() {
-        // If there is work currently being sent or we have no pending work return early
-        if (this.current.size > 0 || this.pending.size === 0) {
+        if (this.pending.size === 0) {
             return;
         }
 
-        // Swap current and pending
-        const temp = this.current;
-        this.current = this.pending;
-        this.pending = temp;
-        const batchOffset = this.pendingOffset;
+        // Invoke the next send after a setImmediate to give IO time to create more batches
+        setImmediate(() => {
```

Comment:
Could you save the `setImmediate` timer id to a property and make it so `sendPending` does nothing if there's an existing timer? Then in this callback, clear that timer id before calling `sendPending` again.

Otherwise this call would keep queueing up more of these callbacks. There's no need to queue up more than one of these since the callback will process/send the whole batch.

Reply:
@GaryWilber I fix it with an **isMessageSending** flag,  is this a graceful solution？

I try to fix it with timer id and **clearImmediate**, but I have no idea to determine whether a timer is cleared.

Reply:
Yea looks like this will work.